lib/appmetrics-prometheus.js

```diff
@@ -273,6 +262,267 @@ function stringHttpRequestDuration(httpURLDataList) {
+    `process_uptime_count_seconds ${process.uptime()}\n`,
+  ];
+}
+
+function stringLoopData(latestLoopEvent) {
+  if (!latestLoopEvent) return;
+  return [
+    '# HELP event_loop_tick_min_milliseconds The shortest tick time in the event loop samples, in milliseconds',
+    '# TYPE event_loop_tick_min_milliseconds guage',
+    `event_loop_tick_min_milliseconds ${latestLoopEvent.minimum}`,
+
+    '# HELP event_loop_tick_max_milliseconds The longest tick time in the event loop samples, in milliseconds',
+    '# TYPE event_loop_tick_max_milliseconds guage',
+    `event_loop_tick_max_milliseconds ${latestLoopEvent.maximum}`,
+
+    '# HELP event_loop_tick_count The number of event loop ticks in the last interval',
+    '# TYPE event_loop_tick_count counter',
+    `event_loop_tick_count ${latestLoopEvent.count}`,
+
+    '# HELP event_loop_tick_average_milliseconds The average tick time in the event loop samples, in milliseconds',
+    '# TYPE event_loop_tick_average_milliseconds guage',
+    `event_loop_tick_average_milliseconds ${latestLoopEvent.average}`,
+
+    '# HELP event_loop_cpu_user The percentage of 1 CPU used by the event loop thread in user code the last interval. This is a value between 0.0 and 1.0.',
+    '# TYPE event_loop_cpu_user guage',
+    `event_loop_cpu_user ${latestLoopEvent.cpu_user}`,
+
+    '# HELP event_loop_cpu_system The percentage of 1 CPU used by the event loop thread in system code in the last interval. This is a value between 0.0 and 1.0.',
+    '# TYPE event_loop_cpu_system guage',
+    `event_loop_cpu_system ${latestLoopEvent.cpu_system}\n`,
+  ].join('\n');
+}
+
+function stringGCData(latestGCEvent, maxHeapUsed, gcDurationTotal) {
+  if (!latestGCEvent) return;
+  return [
+    '# HELP heap_size_bytes The size of the JavaScript heap in bytes',
+    '# TYPE heap_size_bytes guage',
+    `heap_size_bytes ${latestGCEvent.size}`,
+
+    '# HELP heap_memory_used_bytes The amount of memory used on the JavaScript heap in bytes',
+    '# TYPE heap_memory_used_bytes guage',
+    `heap_memory_used_bytes ${latestGCEvent.used}`,
+
+    '# HELP heap_memory_used_max_bytes The maximum amount of memory used on the JavaScript heap in bytes',
+    '# TYPE heap_memory_used_max_bytes count',
+    `heap_memory_used_max_bytes ${maxHeapUsed}`,
+
+    '# HELP gc_cycle_duration_milliseconds The duration of the GC cycle in milliseconds',
+    '# TYPE gc_cycle_duration_milliseconds guage',
+    `gc_cycle_duration_milliseconds ${latestGCEvent.duration}`,
+
+    '# HELP gc_cycle_duration_total_milliseconds The total duration of all GC cycles in milliseconds',
+    '# TYPE gc_cycle_duration_total_milliseconds count',
+    `gc_cycle_duration_total_milliseconds ${gcDurationTotal}\n`,
+  ].join('\n');
+}
+
+function stringHttpRequestsTotal(aggregateHttpEvent) {
+  if (isEmptyObject(aggregateHttpEvent)) return;
+  return [
+    '# HELP http_requests_snapshot_total Total number of HTTP requests received in this snapshot.',
```

Comment:
what's a snapshot? how is this not just a derived value of `http_requests_total`?